spring-cloud-sleuth-zipkin-stream/src/main/java/org/springframework/cloud/sleuth/zipkin/stream/SamplingZipkinSpanIterator.java

```diff
@@ -138,4 +146,43 @@ public void remove() {
 		}
 		return zipkinSpan.build();
 	}
+
+	private static void addLocalComponentAnnotation(Span span, Builder zipkinSpan,
```

Comment:
nit: `ensureLocalComponent` as that allows you to completely dodge Annotation vs BinaryAnnotation (latter being what's in question here)